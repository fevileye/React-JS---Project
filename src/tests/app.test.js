import React from 'react';
import App from '../app.js';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('App Component', () => {
    it('Current Currency Start with 1', () => {
        const wrapper = shallow(<App />)
        const currentCurrencyState= wrapper.state().currentCurrency;
        expect(currentCurrencyState).toEqual(1);
    });

    it('Selected Rate Start With IDR,EUR,GBP,SGD', () => {
        const wrapper = shallow(<App />)
        const selectedRate= wrapper.state().selectedRate;
        expect(selectedRate).toEqual(['IDR','EUR','GBP','SGD']);
    });

    it('Available List Down IDR,EUR,GBP,SGD', () => {
        const wrapper = shallow(<App />)
        const availableListDown= wrapper.state().availableListDown;
        expect(availableListDown).toEqual(['-- Select --','CAD','GBP','CHF','SGD','INR','MYR','JPY','KRW','EUR','IDR']);
    });

    it ('matches the snapshot of App', () =>{
        const tree= renderer.create(<App />).toJSON();

        expect(tree).toMatchSnapshot();
    })

});