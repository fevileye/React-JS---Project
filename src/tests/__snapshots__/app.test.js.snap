// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot of App 1`] = `
<div
  className="mainApp"
>
  <header
    className="MuiPaper-root-010 MuiPaper-elevation4-016 MuiAppBar-root-01 MuiAppBar-positionStatic-05 MuiAppBar-colorPrimary-08"
  >
    <div
      className="MuiToolbar-root-037 MuiToolbar-regular-039 MuiToolbar-gutters-038"
    >
      <button
        aria-label="Open drawer"
        className="MuiButtonBase-root-047 MuiIconButton-root-041 MuiIconButton-colorInherit-042"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-046"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-050"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-057"
        />
      </button>
      <div
        className="baseCurrencyDetail"
      >
        <h2
          className="MuiTypography-root-064 MuiTypography-title-070 MuiTypography-colorInherit-083 MuiTypography-noWrap-080"
        >
          <p
            id="Title"
          >
            United States Dollar
          </p>
        </h2>
      </div>
      <div
        className="baseCurrency"
      >
        <h2
          className="MuiTypography-root-064 MuiTypography-title-070 MuiTypography-colorInherit-083 MuiTypography-noWrap-080"
        >
          <p
            id="TitleCurrency"
          >
            USD
          </p>
        </h2>
      </div>
      <div
        className="currencyInput"
      >
        <div
          className="MuiInput-root-089"
        >
          <input
            className="MuiInput-input-097 MuiInput-inputType-0100"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Currency"
            type="number"
            value={1}
          />
        </div>
      </div>
    </div>
  </header>
  <div
    className="addButton"
  >
    <button
      aria-describedby={null}
      aria-label="Add"
      className="MuiButtonBase-root-047 MuiButton-root-0109 MuiButton-fab-0126 MuiButton-contained-0120 MuiButton-containedPrimary-0121 MuiButton-raised-0123 MuiButton-raisedPrimary-0124"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      title="Add"
      type="button"
    >
      <span
        className="MuiButton-label-0110"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-050"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-057"
      />
    </button>
  </div>
  <div
    className="currency-list"
  />
  
</div>
`;
